dll/win32/shell32/utils.cpp

```diff
@@ -28,6 +28,19 @@ static BOOL OpenEffectiveToken(DWORD DesiredAccess, HANDLE *phToken)
     return ret;
 }
 
+/*************************************************************************
+ *                ShortSizeFormatW (SHELL32.204)
+ */
+EXTERN_C
+LPWSTR WINAPI
+ShortSizeFormatW(
+    _In_ DWORD dwNumber,
+    _Out_writes_z_(0x8FFF) LPWSTR pszBuffer)
```

Comment:
Where does the magical "0x8FFF" come from? I mean, why isn't it 0x9000 or some other size?

Reply:
> Where does the magical "0x8FFF" come from?

Very strange. The output is always going to be pretty short, "1023 bytes", "5.32 GB" etc. `0x8FFF` makes no sense, it is not a UNICODE_STRING limit or any other kind of limit I have ever heard of.

Reply:
@katahiromz Why can't you explain where this limit comes from and why it needs to be this large?

Reply:
The size unit depends on locale info.
The system locale info can be hacked.
So it is unpredictable.

The original implementation uses this value.